python/pyarrow/array.pxi

```diff
@@ -1475,6 +1475,22 @@ cdef class Array(_PandasConvertible):
         return self.take(indices)
 
     def _to_pandas(self, options, types_mapper=None, **kwargs):
+        pandas_dtype = None
+
+        if types_mapper:
+            pandas_dtype = types_mapper(self.type)
+        elif self.type.id == _Type_EXTENSION:
+            try:
+                pandas_dtype = self.type.to_pandas_dtype()
+            except NotImplementedError:
+                pass
+
+        # Only call __from_arrow__ for Arrow extension types or when explicitly
+        # overridden via types_mapper
+        if hasattr(pandas_dtype, '__from_arrow__'):
+            arr = pandas_dtype.__from_arrow__(self)
+            return pandas_api.series(arr, copy=False)
```

Comment:
Given that this code is both here and in the ChunkedArray version, and both methods also end up calling `_array_like_to_pandas`, we can maybe move this logic there to avoid duplicating it.

Reply:
+1 on code deduplication, I'll give it a shot. The only differences currently are the parameters passed into `pandas_api.series(arr)`. Arrays want copy=False, and ChunkedArray has enough info to add name=self._name.

Reply:
From my testing, `Array` does always have a `None` "column" name. Good to know `ChunkedArray` could benefit from `copy=False`! I thought maybe it was left out on purpose since the memory is not guaranteed to be contiguous.

Reply:
> I thought maybe it was left out on purpose since the memory is not guaranteed to be contiguous.

ChunkedArray itself can indeed consist of multiple chunks, but at the moment we reach the `pandas_api.series(arr, copy=False)` code, we already converted the ChunkedArray to a single contiguous numpy array (or pandas ExtensionArray). And anyway, this `copy=False` is mostly to indicate to pandas they don't need to do an extra copy (https://github.com/apache/arrow/pull/34593).